src/command-line/uninstall.js

```diff
@@ -40,63 +46,48 @@ program
 			process.exit(1);
 		};
 
-		// First, we check if the package is installed with `npm list`
-		const list = child.spawn(
-			npm,
+		const yarn = path.join(
+			__dirname,
+			"..",
+			"..",
+			"node_modules",
+			"yarn",
+			"bin",
+			"yarn.js"
+		);
+
+		let success = false;
+		const remove = child.spawn(
+			process.execPath,
 			[
-				"list",
-				"--depth",
-				"0",
-				"--prefix",
+				yarn,
+				"remove",
+				"--json",
+				"--ignore-scripts",
+				"--non-interactive",
+				"--cwd",
 				packagesParent,
 				packageName,
-			],
-			{
-				// This is the same as `"inherit"` except:
-				// - `process.stdout` is piped so we can test if the output mentions the
-				//   package was found
-				// - `process.stderr` is ignored to silence `npm ERR! extraneous` errors
-				stdio: [process.stdin, "pipe", "ignore"],
-			}
+			]
 		);
 
-		list.stdout.on("data", (data) => {
-			// If the package name does not appear in stdout, it means it was not
-			// installed. We cannot rely on exit code because `npm ERR! extraneous`
-			// causes a status of 1 even if package exists.
-			if (!data.toString().includes(packageName)) {
-				log.warn(packageWasNotInstalled);
-				process.exit(1);
-			}
-		});
+		remove.stdout.on("data", (data) => {
+			data.toString().trim().split("\n").forEach((line) => {
+				line = JSON.parse(line);
 
-		list.on("error", errorHandler);
-
-		list.on("close", () => {
-			// If we get there, it means the package exists, so uninstall
-			const uninstall = child.spawn(
-				npm,
-				[
-					"uninstall",
-					"--no-progress",
-					"--prefix",
-					packagesParent,
-					packageName,
-				],
-				{
-					// This is the same as `"inherit"` except `process.stdout` is silenced
-					stdio: [process.stdin, "ignore", process.stderr],
+				if (line.type === "success") {
+					success = true;
 				}
-			);
+			});
+		});
 
-			uninstall.on("error", errorHandler);
+		remove.on("error", errorHandler);
 
-			uninstall.on("close", (code) => {
-				if (code !== 0) {
-					errorHandler(code);
-				}
+		remove.on("close", (code) => {
+			if (!success || code !== 0) {
+				errorHandler(code);
+			}
 
-				log.info(`${colors.green(packageName)} has been successfully uninstalled.`);
-			});
+			log.info(`${colors.green(packageName)} has been successfully uninstalled.`);
```

Comment:
Will this run even if there was an error? We shouldn't do that.

Reply:
errorHandler exits the process.

Reply:
Oh duh, yeah. I don't like that, but let's not change it as part of this PR.

Reply:
I added a return to make it clear.

Reply:
@YaManicKill, I'm confused, probably looking at a wrong line or something, but what don't you like?

Reply:
I'm not a fan of the errorHandler exiting the system. But that's not part of this PR.